sql/updates/reverse-dev.sql

```diff
@@ -103,3 +103,24 @@ DROP FUNCTION IF EXISTS _timescaledb_functions.uuid_version;
 
 DELETE FROM _timescaledb_catalog.compression_algorithm WHERE id = 7 AND version = 1 AND name = 'COMPRESSION_ALGORITHM_UUID';
 
+-- block downgrade if a table has NULL orderby setting (not allowed in 2.21)
+DO $$
+BEGIN
+  IF EXISTS (
+        SELECT 1
+        FROM _timescaledb_catalog.compression_settings
+        WHERE orderby IS NULL
+        ) THEN
+    RAISE EXCEPTION 'TimescaleDB 2.21 and below can not have null hypertable orderby settings. Alter table to set them.';
+  END IF;
+END
+$$;
+
+-- remove empty segmentby
+DO $$
+BEGIN
+  UPDATE _timescaledb_catalog.compression_settings
+  SET segmentby = NULL
+  WHERE segmentby = '{}';
+END;
+$$ LANGUAGE plpgsql;
```

Comment:
```suggestion
-- convert empty array segmentby to NULL
UPDATE _timescaledb_catalog.compression_settings
SET segmentby = NULL
WHERE segmentby = '{}';
```
No need for this to be in DO block, right?